gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenDynamicPartitionPruningSuite.scala

```diff
@@ -281,6 +282,38 @@ abstract class GlutenDynamicPartitionPruningSuiteBase
     }
   }
 
+  test(GLUTEN_TEST + "SPARK-38674: Remove useless deduplicate in SubqueryBroadcastExec") {
+    withTable("duplicate_keys") {
+      withSQLConf(SQLConf.DYNAMIC_PARTITION_PRUNING_ENABLED.key -> "true") {
+        Seq[(Int, String)]((1, "NL"), (1, "NL"), (3, "US"), (3, "US"), (3, "US"))
+          .toDF("store_id", "country")
+          .write
+          .format(tableFormat)
+          .saveAsTable("duplicate_keys")
+
+        val df = sql("""
+                       |SELECT date_id, product_id FROM fact_sk f
+                       |JOIN duplicate_keys s
+                       |ON f.store_id = s.store_id WHERE s.country = 'US' AND date_id > 1050
+          """.stripMargin)
+
+        checkPartitionPruningPredicate(df, withSubquery = false, withBroadcast = true)
+
+        val subqueryBroadcastExecs = collectWithSubqueries(df.queryExecution.executedPlan) {
+          case s: ColumnarSubqueryBroadcastExec => s
+        }
+        assert(subqueryBroadcastExecs.size === 1)
+        // Not relevant. See MetricsApiImpl.genColumnarSubqueryBroadcastMetrics
+        // and org.apache.spark.sql.execution.ColumnarSubqueryBroadcastExec.metrics
```

Comment:
This metrics got added in spark 3.4 in SPARK-38674, we should make changes in gluten to add this metric.

Reply:
Done